server/sonar-server/src/main/java/org/sonar/server/notification/ws/AddAction.java

```diff
@@ -124,8 +124,9 @@ private void add(AddRequest request) {
     try (DbSession dbSession = dbClient.openSession(false)) {
       checkPermissions(request);
       UserDto user = getUser(dbSession, request);
-      ComponentDto project = searchProject(dbSession, request);
-      notificationUpdater.add(dbSession, request.getChannel(), request.getType(), user, project);
+      Optional<ComponentDto> project = searchProject(dbSession, request);
+      project.ifPresent(p -> userSession.checkComponentPermission(UserRole.USER, p));
```

Comment:
since we already perform checks on the project there, I'd put this in searchProject for consistency